string init
float64[] pos